drivers/LVHDoHBASR.py

```diff
@@ -195,13 +195,25 @@ def detach(self, sr_uuid):
             except:
                 pass
 
+    def _remove_device_nodes(self):
+        """
+        Remove the kernel device nodes
+        """
+        nodes = glob.glob('/dev/disk/by-scsid/%s/*' % self.SCSIid)
+        util.SMlog('Remove_nodes, nodes are %s' % nodes)
+        for node in nodes:
+            with open('/sys/block/%s/device/delete' %
+                      (os.path.basename(node)), 'w') as f:
+                f.write('1\n')
+
     def delete(self, sr_uuid):
         self._pathrefresh(LVHDoHBASR)
         try:
             LVHDSR.LVHDSR.delete(self, sr_uuid)
         finally:
             if self.mpath == "true":
                 self.mpathmodule.reset(self.SCSIid, explicit_unmap=True)
+            self._remove_device_nodes()
```

Comment:
Could this fail and leave the SR in an unexpected state?

Reply:
No, it either succeeds or leaves things as they are today.